code/obj/machinery/manufacturer.dm

```diff
@@ -648,6 +676,89 @@
 					src.pulse(twire)
 				src.build_icon()
 
+			if (href_list["card"])
+				if (src.scan) src.scan = null
+				else
+					var/obj/item/I = usr.equipped()
+					if (istype(I, /obj/item/card/id))
```

Comment:
might be nice to let IDs in pdas work, and to put this on an attack_by override as well, so you can just smack the fab with your id

Reply:
Maybe we should get ourselves an access carrying component. IDs would have it by default and they'd transfer it into the PDA when inserted. That would make stuff like this cleaner.

Reply:
All this stuff (Id in PDA, etc) has a proc to check/receive it

Reply:
Added PDA check to both the attackby and scan link in pending commit